lessons/misc/webdev-intro/contents.md

```diff
@@ -0,0 +1,562 @@
+    console.log('object', this);
+    console.log('arg1', arg1);
+    console.log('arg2', arg2);
+}
+var foo = {name: 'bar', method: f}
+```
+
+The following are equivalent
+
+```js
+foo.method('hi');
+f.bind(foo)('hi');
+f.apply(foo, ['hi');
+f.call(foo, 'hi');
+```
+
+## Conditional statements
+
+The if/else if/else statement is same as in C
+
+```js
+if (condition) {
+    codeBlock;
+}
+else if (condition2)
+    statement;
+else {
+    codeBlock;
+}
+```
+
+There is also `switch..case`, which we'll see in this app's code.
+
+## Casting in comparison
+
+There are two equalities in JavaScript
+ - `==` or `!=` perform casting when comparing values so `12 == true` is `true`
+ - `===` or `!==` don't cast, so `1 === true` is `false
+
+Casting also happens for `<`, `<=`, etc. 
+
+A logical "not" is the operator `!`, which does cast to booleans. So you can do `!!undefined` to get `true`.
+
+## Loops
+
+There are `while`, `do...while`, and `for` loops. All follow classic C syntax, but there are special forms of `for` loops
+
+```js
+var shoppingList = ['banana', 'apple', 'bread'];
+shoppingList.title = 'My Groceries';
+
+/* This will print the object's properties */
+for (let i in shoppingList)
+    console.log(i);
+
+/* and this prints the elements in the _iterable_ object */
+for (let x of shoppingList)
+    console.log(x);
+
+/* for arrays you could still do */
+for (let i=0; i<shoppingList.length; i++) {
+    console.log(shoppingList[i]);
+}
+```
+
+The `for...of` syntax behaves most like Python's `for...in` loop.
+
+## Interacting with the DOM
+
+Go over the code for this "app".
+
+Note:
+- Not organized well ("global" variables, weird scope, etc.)
+- `Element.getElementById` and `.getElementsByClassName`
+- `classList.add/remove()`
+- slide creation and navigation bar element creation
+- this [link](#10) should go to slide 10. 
+
+
+## Classes/Objects
+
+JavaScript is *prototype*-based. This is different from *class*-based, but has similar functionality.
+
+- An Object is like a key-value store,
+- A constructor is a function applied to a prototype object
+    - `var obj = new MyClass();`
+- methods/properties can be through the prototype
+    - `MyClass.prototype = {foo: function() { return 'foo';}, bar: 'bar'}`
+- or just `ownProperty`
+    - `function MyClass() { this.asdf = function() {return this.bar;}`
+- properties go up the property chain
+    - `obj.toString()` would go to `Object.prototype.toString.apply(obj)`
+
+## Prototype example
+
+```js
+function A() {
+    this.varA = 'a';
+}
+
+function B() {
+    A.call(this);
+    this.varB = 'b';
+}
+
+A.prototype = { foo: 1, bar: function(x) { return this.foo + 1;}};
+B.prototype = Object.create(A.prototype);
+B.prototype.foo = 3;
+
+a = new A();
+b = new B();
+
+a.bar(1);
+b.bar(1);
+
+B.prototype.foo = -1;
+b.bar(1);
+```
+
+## Class syntax
+
+Just a syntactic wrapping of what we covered before.
+
+```js
+
+class C extends B {
+    constructor(c) {
+        super();
+        this.foo = -1
```

Comment:
Missing semicolon (yea, yea, optional semicolons, etc., not the place for that discussion.)